blog/writeFrontMatterCache.test.ts

```diff
@@ -0,0 +1,21 @@
+import { validateFrontMatter } from "./writeFrontMatterCache";
+
+describe("writeFrontMatterCache", () => {
```

Comment:
this was the first set of tests i was working on writing the other day. my initial thought to seeing this version is that the fact that `validateFrontMatter` and `buildFrontMatterCache` are exported but _only_ used in tests is a code smell in my opinion. but that's more of a philosophical thought and i'd rather not bog you down with testing philosophy back-and-forth (especially since the bigger picture is that we understandably don't have a project-wide testing style here). 

so i'm going to hold off on writing tests around this and instead i now propose we focus on getting this merged asap. and for testing, i'm going to follow this up with a larger effort to add more testing to this project as a whole and analyzing the tests that are already there. 

thoughts on that strategy? @brookslybrand?

Reply:
maybe a better way to ask that question is what are the dealbreakers preventing this PR from getting merged? i normally would say testing, but i'm proposing hold off for a bit cuz i'm going to make sure we fast follow.

Reply:
The reason for those exports is just to use them in testing, and pretty much no other reason. I did that for familiarity's sake because I've never in my career had to write tests that check the file system and assert on a file's expected output. It feels weird to me, but maybe that's just my front-end bias speaking. If you think that's a valid test to write, I would say you can adjust this as needed.

Reply:
I'm personally good with having tests before merging and getting good coverage. This is a learning experience for me and TBH I miss making stuff with others that feels high quality and airtight.

Reply:
I might not be understanding right, but Colby are you saying your concerned that these 2 functions, which are used inside of `writeFrontMatterCache`, are exported just so they can be tested?

Personally I think this is fine if you want to test them, however I would say it's more valuable to just test `writeFrontMatterCache` and not worry about the implementation details. Your test probably shouldn't care about `validateFrontMatter` or any other implementation details of `writeFrontMatterCache`, just that `writeFrontMatterCache` is doing the right thing in all situations.

That's my opinion, but yeah, I don't think it's a big deal or a blocker. I do like the idea of following up with further testing holistically and improving coverage a bit.